protocols/gossipsub/src/behaviour.rs

```diff
@@ -1226,19 +1226,19 @@ where
             }
 
             for id in ids {
-                if !self.duplicate_cache.contains(&id) && !self.pending_iwant_msgs.contains(&id) {
-                    if self
+                if !self.duplicate_cache.contains(&id)
+                    && !self.pending_iwant_msgs.contains(&id)
+                    && self
                         .peer_score
                         .as_ref()
                         .map(|(_, _, _, promises)| !promises.contains(&id))
                         .unwrap_or(true)
```

Comment:
This is quite the monster boolean condition now. Perhaps we should extract 3 boolean variables that we can then chain together to make this more readable?

Reply:
I guess functions would allow for readability and short-circuiting?

Reply:
Indeed, a lambda would do the trick! For the first two conditions I don't think it would help much, i.e. `!self.duplicate_cache.contains(&id)` would most likely become `duplicate_cache.contains(&id)`. Is it fine for you?

The last one is definitely a monster though and I'd be happy to turn it into a short lambda. Any suggestions on a meaningful name for it?

Reply:
Probably even better would be to do:

```rust
impl Gossipsub {
	fn want_message(&self, id: ???) -> bool {
		if self.duplicate_cache.contains(&id) {
			return false;
		}

		if self.pending_iwant_msgs.contains(&id) {
			return false;
		}

		self.peer_score
     		  .as_ref()
            .map(|(_, _, _, promises)| !promises.contains(&id))
            .unwrap_or(true)
	}
}
```

Although I think clippy might complain about this one being possible to write as a `&&` chain.

Reply:
Good idea! I went ahead with similar change. I was a bit confused with a `contains` method being mutable for no visible reason so I made them immutable. Maybe pedantic clippy would have caught that before.